src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,84 @@
+import { Dispatch, SetStateAction } from 'react';
+import { Progress } from '../../types/Progress';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  itemsLeft: number,
+  setProgress: (type: Progress) => void;
+  isAnyCompleted: boolean,
+  todos: Todo[],
+  onClear: (todoId: number) => void,
+  setGlobalLoading: Dispatch<SetStateAction<boolean>>,
+}
+
+export const Footer:React.FC<Props> = ({
+  itemsLeft,
+  setProgress,
+  isAnyCompleted,
+  todos,
+  onClear,
+  setGlobalLoading,
+}) => {
+  const ClearAllCompleted = async () => {
+    setGlobalLoading(true);
```

Comment:
You should show loaders only for todos that you plan to update or delete. Because now I see a loader on all todos, when only a few will be deleted

As you need to show multiple loaders at the same time you can store in the state an array of ids of todos that currently loading. So in the todo component, you can check if the current todo id is in the array, and show the loader.